### PR TITLE
doc Refer to the Streams section from TCP stream

### LINKED ISSUE #21308: unclear tcp stream documentation
I think the tcp stream documentation is lacking some basic explanations. Although I'm very familiar with the java socket API, it took me quite a while to understand how the API is supposed to work and to set up a simple working TCP stream based example. The main reason is IMHO that some basic concepts are not explained in the documentation. They are probably explained in other places (e.g. presentations), but as a user I would like to get started quickly by only reading the documentation.

Concretely, I'd like to add a short paragraph that shows how the flows are related to the underlying socket's input and output streams. Also, I'd like to make it very explicit that the stream implementation does not add any additional bytes to the data that is sent over the socket.

I can imagine that these can be derived from what is already there, but if you're new to streams and don't have the time to carefully read everything top-to-bottom then this is far from obvious.
 
PS: I'm happy to make a PR

### PR DESCRIPTION
**Pull request by richardimaoka**
Fixes #21308

Maybe not necessary to mention Akka HTTP from this TCP section. However, since Akka HTTP is well documented with lot more examples, I thought users get hints from the Akka HTTP section.

[Java TCP Stream IO doc](http://doc.akka.io/docs/akka/2.4/java/stream/stream-io.html)
[Scala TCP Stream IO doc](http://doc.akka.io/docs/akka/2.4/scala/stream/stream-io.html)
### After this change

image

### CODE REVIEW

File: akka-docs/rst/java/stream/stream-io.rst [hunk 1]
```diff
@@ -11,6 +11,14 @@ as the library does it transparently for you.
 
 .. _Actor based TCP handling: http://doc.akka.io/docs/akka/current/java/io-tcp.html
 
+.. note::
+  If you are not familiar with Akka Streams basic concepts, like ``Source``, ``Sink`` and ``Flow``,
+  please refer to the `Streams section`_ of the document. Also higher level APIs like ``bind`` are
+  used in Akka HTTP too. So you may get some hints from the `Akka HTTP introduction section`_.
+
+.. _Streams section: http://doc.akka.io/docs/akka/current/scala/stream/index.html
```
Comment (patriknw) [inline akka-docs/rst/java/stream/stream-io.rst]: this should be a cross link, search for similar `:ref:`

**Comment by drewhk** [comment]
LGTM

**Comment by richard-imaoka** [comment]
Sorry still working on this to add diagrams + reflect comments from Patrik. Will come back to this in a few days.

File: akka-docs/rst/java/stream/stream-io.rst (hunk 1 above)
Comment (ktoso) [inline akka-docs/rst/java/stream/stream-io.rst]: +1

**Review by ktoso — CHANGES_REQUESTED** [review]
LGTM, after ref fixed

**Comment by richard-imaoka** [comment]
Added images. 

Ping @BertRobben in case he has comments on the images.
image

image

File: akka-docs/rst/java/stream/stream-io.rst (hunk 1 above)
Comment (richard-imaoka) [inline akka-docs/rst/java/stream/stream-io.rst]: Fixed this.

**Comment by BertRobben** [comment]
The images look great! Nice job.Now it should be much clearer for starters.

thanks,

Bert

**Comment by patriknw** [comment]
LGTM

**Pull request merged**